test/src/org/labkey/test/components/cds/ActiveFilterDialog.java

```diff
@@ -0,0 +1,153 @@
+package org.labkey.test.components.cds;
+
+import org.jetbrains.annotations.Nullable;
+import org.junit.Assert;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.components.html.Input;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class ActiveFilterDialog extends WebDriverComponent<ActiveFilterDialog.ElementCache>
+{
+    private final WebElement _activeFilterDialogEl;
+    private final WebDriverWrapper _webDriverWrapper;
+
+    public ActiveFilterDialog(BaseWebDriverTest test)
+    {
+        _webDriverWrapper = test;
+        _activeFilterDialogEl = newElementCache().saveGroupEl.refindWhenNeeded(_webDriverWrapper.getDriver());
+    }
+
+    public void waitForSaveGroupDialog()
+    {
+        _webDriverWrapper.waitForElement(elementCache().saveGroupEl);
+    }
```

Comment:
This is all just a bit nonstandard.
The element cache shouldn't contain the locator for the component; waiting for the element should happen automatically; the constructor should take a `WebDriver` not a test.
```suggestion
    private final WebElement _activeFilterDialogEl;
    private final WebDriver _driver;

    public ActiveFilterDialog(WebDriver driver)
    {
        _driver = driver;
        _activeFilterDialogEl = Locator.tagWithId("div", "filterstatus-id").waitForElement(_webDriverWrapper.getDriver());
    }
```
If you use the `WebDriverTestComponent` you get the standard component pattern for free.

Reply:
Done